### DISCUSSION TITLE
How to handle multiples devices with the same `manufacturerName`? (Tuya)

### DISCUSSION THREAD

**Discussion by etiennedub** [Koenkk/zigbee2mqtt#21935] [Devices]
Hi, I have a Tuya curtain motor label as `LY-1668` but it's recognized as `LY-108` because they share the same `modelID` and `manufacturerName`. I would like to expose new tuyaDatapoints but without impacting the existing `LY-108`. Is there any way to use another attribute to filter the device?

Related issue: https://github.com/Koenkk/zigbee2mqtt/issues/21876

**Comment by schauveau** [comment]
I believe that for non-Tuya devices, the matching is done in isFingerprintMatch
 https://github.com/Koenkk/zigbee-herdsman-converters/blob/7ecea3b807ec7643493f94614e84d0343423267d/src/index.ts#L322
All the attributes tested in that function should work.

A few interesting examples:

- match `ieeeAddr`   

https://github.com/Koenkk/zigbee-herdsman-converters/blob/7ecea3b807ec7643493f94614e84d0343423267d/src/devices/sunricher.ts#L332

remark: isFingerprintMatch is calling `device.ieeeAddr.match()` so a regex is required.  To match a single `ieeeAddr`, use something like `/^0x1234567812345678$/`

- match multiple values of `applicationVersion`. That one also sets `priority:-1` probably to have precedence over the previous definitions.
  
https://github.com/Koenkk/zigbee-herdsman-converters/blob/7ecea3b807ec7643493f94614e84d0343423267d/src/devices/tuya.ts#L3796

**Comment by Koenkk** [comment]
whats the datapoint? maybe its also supported by the other device
